src/microbe_stage/biomes.cpp

```diff
@@ -85,3 +138,54 @@ CScriptArray*
         (compounds | boost::adaptors::map_keys).end(),
         Leviathan::ScriptExecutor::Get()->GetASEngine(), "array<uint64>");
 }
+
+ChunkData*
+    Biome::getChunk(size_t type)
+{
+    return &chunks[type];
+}
+
+CScriptArray*
+    Biome::getChunkKeys() const
+{
+    return Leviathan::ConvertIteratorToASArray(
+        (chunks | boost::adaptors::map_keys).begin(),
+        (chunks | boost::adaptors::map_keys).end(),
+        Leviathan::ScriptExecutor::Get()->GetASEngine(), "array<uint64>");
+}
+
+// ----- //
+
+ChunkCompoundData*
+    ChunkData::getCompound(size_t type)
+{
+    return &chunkCompounds[type];
+}
+
+CScriptArray*
+    ChunkData::getCompoundKeys() const
+{
+    return Leviathan::ConvertIteratorToASArray(
+        (chunkCompounds | boost::adaptors::map_keys).begin(),
+        (chunkCompounds | boost::adaptors::map_keys).end(),
+        Leviathan::ScriptExecutor::Get()->GetASEngine(), "array<uint64>");
+}
+
+int
+    ChunkData::getMeshListSize()
+{
+    return this->meshes.size();
+}
+
+std::string
+    ChunkData::getMesh(int index)
+{
+    // Some error checking
+    if(index >= 0 && index < this->meshes.size()) {
+        return this->meshes.at(index);
+    } else {
+        LOG_ERROR(
+            "Mesh at index " + std::to_string(index) + " does not exist!");
+        return "error";
```

Comment:
This could be made to throw an exception. That safely aborts the current running script with the error (if this uses Leviathan::InvalidArgument exception)

Reply:
Im keeping track these, this is now done on my end